cli/src/commands/version.rs

```diff
@@ -0,0 +1,39 @@
+use std::process::Command;
+
+use clap::Parser as ClapParser;
+
+/// Prints version information
+///
+/// This subcommand is implemented for backward compatibility reasons.
+/// Its behavior should be the same as that of the `--version` flag which
+/// is handled internally by clap.
+#[derive(Debug, ClapParser)]
+pub struct Opts {}
+
+pub fn run(_opts: Opts) {
+    let commit = match Command::new("git")
+        .args(["rev-parse", "--short=11", "HEAD"])
+        .output()
+    {
+        Ok(o) if o.status.success() => {
+            let sha = String::from_utf8_lossy(&o.stdout).trim().to_owned();
+            sha
+        }
+        Ok(o) => {
+            println!("cargo:warning=Git command failed with status: {}", o.status);
```

Comment:
these git errors are intended to be compile warnings as these are copy-pasted from a build script...

we can keep them but I think these would be better suited to be logged instead of the "ordinary" CLI output as it might be confusing.

Reply:
Done